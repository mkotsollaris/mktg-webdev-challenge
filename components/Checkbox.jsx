import React from 'react';

const Checkbox = ({ text, checked }) => {
    return <div>
        {checked ? <div>✔️</div> : null}
        {!text ? null : <div>
            {text}
        </div>}
    </div>
}

export default Checkbox;