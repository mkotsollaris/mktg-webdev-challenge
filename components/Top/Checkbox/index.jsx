import { useState } from 'react'
import style from './style.module.css'

const Checkbox3 = ({
  checked,
  spanText = 'Hide people missing a profile image',
  ...props
}) => {
  // const [isChecked, setIsChecked] = useState();
  const [isFocused, setIsFocused] = useState(false)

  const classNames = isFocused ? ` ${style['on-focus']}` : null

  return (
    <div className={`${style.container}`}>
      <div>
        <div className={style.grid}>
          <input
            className={style.input}
            onFocus={() => {
              setIsFocused(true)
            }}
            onBlur={() => {
              setIsFocused(false)
            }}
            type="checkbox"
            checked={checked}
            {...props}
          />
          <span className={`${style.checkmark} ${classNames}`}></span>
          <span className={style['small-span']}>{spanText}</span>
        </div>
      </div>
    </div>
  )
}

export default Checkbox3
