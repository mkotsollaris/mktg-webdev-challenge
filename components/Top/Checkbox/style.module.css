/* The container */

.container {
  display: block;
  position: relative;
  padding-left: 35px;
  margin-bottom: 12px;
  cursor: pointer;
  font-size: 22px;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.input {
  position: absolute;
  opacity: 0;
  cursor: pointer;
  height: 0;
  width: 0;
}

.on-focus {
  box-shadow: 0 0 0 2px dodgerblue;
}

.checkmark {
  position: absolute;
  top: 0;
  left: 0;
  height: 25px;
  width: 25px;
  background-color: white;
  border: 1px solid dodgerblue;
  border-radius: 2px;

@nest .container:hover input ~ & {
  box-shadow: 0 0 0 2px dodgerblue;
}

@nest .container input:checked ~ & {
  background-color: dodgerblue;
}

&::after {
  content: '';
  position: absolute;
  display: none;
}
}

.container input:checked ~ .checkmark::after {
  display: block;
}

.container .checkmark::after {
  left: 9px;
  top: 5px;
  width: 5px;
  height: 10px;
  border: solid white;
  border-width: 0 3px 3px 0;
  -webkit-transform: rotate(45deg);
  -ms-transform: rotate(45deg);
  transform: rotate(45deg);
}

.small-span {
  font-size: 0.9rem;
  font-weight: lighter;
  color: darkslategray;
}

.grid {
  display: grid;
  grid-template-columns: auto auto;
}
