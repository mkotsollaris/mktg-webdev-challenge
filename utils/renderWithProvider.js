import { render } from '@testing-library/react'
import AppContext from '../context/AppContext'

function renderWithProvider(props, Component) {
  return render(
    <AppContext.Provider value={props}>
      <Component />
    </AppContext.Provider>
  )
}

export default renderWithProvider
