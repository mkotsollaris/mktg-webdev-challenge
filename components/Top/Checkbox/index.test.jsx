import Checkbox from '.'
import renderWithProvider from '../../../utils/renderWithProvider'

it('renders', () => {
  const { asFragment } = renderWithProvider({}, Checkbox)
  expect(asFragment()).toMatchSnapshot()
})
