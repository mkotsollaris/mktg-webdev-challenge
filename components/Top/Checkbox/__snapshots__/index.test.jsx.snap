// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="undefined"
  >
    <div>
      <div>
        <input
          type="checkbox"
        />
        <span
          class="undefined null"
        />
        <span>
          Hide people missing a profile image
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
